openpype/hosts/fusion/plugins/publish/validate_saver_resolution.py

```diff
@@ -0,0 +1,96 @@
+import pyblish.api
+from openpype.pipeline import (
+    PublishValidationError,
+    OptionalPyblishPluginMixin,
+)
+
+from openpype.hosts.fusion.api.action import SelectInvalidAction
+from openpype.hosts.fusion.api import get_current_comp
+
+
+class ValidateSaverResolution(
+    pyblish.api.InstancePlugin, OptionalPyblishPluginMixin
+):
+    """Validate that the saver input resolution matches the projects"""
+
+    order = pyblish.api.ValidatorOrder
+    label = "Validate Saver Resolution"
+    families = ["render"]
+    hosts = ["fusion"]
+    optional = True
+    actions = [SelectInvalidAction]
+
+    @classmethod
+    def get_invalid(cls, instance, wrong_resolution=None):
+        if wrong_resolution is None:
+            wrong_resolution = []
+
+        saver = instance[0]
+        firstFrame = saver.GetAttrs("TOOLNT_Region_Start")[1]
+        comp = get_current_comp()
+
+        # If the current saver hasn't bin rendered its input resolution
+        # hasn't bin saved. To combat this, add an expression in
+        # the comments field to read the resolution
+
+        # Create undo stack so we later can remove all changes
+        comp.StartUndo("Read resolution")
+
+        # Save old comment
+        oldComment = ""
+        hasExpression = False
+        if saver["Comments"][firstFrame] != "":
+            if saver["Comments"].GetExpression() is not None:
```

Comment:
Isn't there a way to get the Image of a tool of which you can directly get its width and height?

I recall implementing a class for Image type in fusion in a library I wrote seven years ago here: https://github.com/Colorbleed/fusionless/blob/master/fusionless/core.py#L1671

I think you can get the value of the input link/parameter to get the image typed value for which you can then do `image.Width`

Which might be `saver.FindMainInput(1)` and then something. If that doesn't work you might need to get the output connected to that input to get its output value `.GetValue`.

I am not behind a PC for a week but I'm pretty sure this can be done through the Python API more directly as opposed to using expressions.

Reply:
From my tests I did to see what ways I could get the resolution, an expression was the only way I could find for a node that hasn't bin rendered yet.
If I read the nodes attributes the resolution variables doesn't exist until I render it.
If I go to the inputs/outputs and read its attributes I only get things like if the input is visible, what index it has and stuff not like. Nothing about the data passing through it.

As the nodes isn't rendered I think an expression is needed to "touch" the node-tree and get the correct resolution calculated. I don't think scripts does that.
I might be wrong but I didn't find any way to get what I needed.